### DISCUSSION TITLE
Public contact form: how to?

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#16237] [Archive]
I read all the discussions about the subject, and looked at the excellent [Directus Roles and Permissions, Common Use Cases](https://www.youtube.com/watch?v=ruzkqxDIG-Y)

What are the steps, beyond?

- create an `Inbox` collection
- a role `Website` with no App Access and just a Create permission on the `Inbox` collection
- a `Website user` that has `Website` role and save its token as `DIRECTUS_TOKEN='xxx'` in `.env`

What do I have to do to get something like the https://directus.io/contact/ build with Directus and make it publicly available?

**Comment by br41nslug** [comment]
What to do next depends highly on what you're using as a frontend https://directus.io/guides/

**Reply by Pachat** [reply]
So you mean that there is presently no way to have it inside Directus even creating a custom extension,?

**Reply by br41nslug** [reply]
You could probably implement such a thing in a custom endpoint sure but they weren't designed to serve HTML.

**Comment by raphourbe** [comment]
Hi @Pachat 

I am currently developing a Saas app. I needed to have a newsletter subscription form (which is not far from the idea of a contact form in the implementation I guess https://directus.io/contact/). I have done a tutorial on my blog: https://rphl.dev/blog/creating-newsletter-subscription-form-directus 

I hope it might give you an idea to reach your goal.

**Comment by rijkvanzanten** [comment]
cc @erondpowell 👨🏻‍🍳

**Reply by erondpowell** [reply]
@raphourbe @Pachat 

There's a new section of the docs coming: [The Directus Cookbook](https://github.com/directus/docs/pull/197)
The merge is at least a few weeks away at present, but it will be a place to find cool recipes like this.

We haven't nailed down all the details yet, but the big hope and dream is to make community contributions a seamless experience.

If you're interested in that and have any thoughts/ideas, feel free to share a comment on the PR.
